java/src/c21415904/SarahVisual.java

```diff
@@ -0,0 +1,249 @@
+package c21415904;
+
+import ie.tudublin.visual.VObject;
+import ie.tudublin.visual.VScene;
+import ie.tudublin.visual.Visual;
+import processing.core.PApplet;
+import processing.core.PVector;
+import ddf.minim.AudioBuffer;
+
+public class SarahVisual extends VScene {
+    Visual v;
+    Clock clock;
+    VObject so1;
+    VObject wf;
+    VObject cwf;
+    VObject hwf;
+    VObject ecwf;
+    VObject sp;
+    VObject td;
+
+    AudioBuffer ab;
+    
+
+    public SarahVisual(Visual v) {
+        super(v);
+        this.v = v;
+
+        ab = v.audioPlayer().mix;
+
+        clock = new Clock(v, new PVector(0, 0, 0));
+        so1 = new SphereOrbit(v, new PVector(0, 0, 0));
+        wf = new WaveForm(v, new PVector(0, 0, 0));  
+        cwf = new CircleWF(v, new PVector(v.width/2, v.height/2, 0));      
+        hwf = new Hex(v, new PVector(v.width/2, v.height/2, 0));  
+        td = new twoDist(v, new PVector(0, 0, 0));     
+    }
+
+    public void render(int elapsed) {
+        // 1:48 - 2:30 - Instrumental
+        if (elapsed > v.toMs(0, 0, 0) && elapsed < v.toMs(1, 2, 0)) {
```

Comment:
Later on you can use the Animations class to decide when to render the stuff, it will allow you to make fading transitions 
```suggestion
        if (Math.round(effect.getValue) != 0) {
```